gems/pending/spec/spec_helper.rb

```diff
@@ -32,3 +36,30 @@
   config.backtrace_exclusion_patterns << %r{/lib\d*/ruby/[0-9]}
   config.backtrace_exclusion_patterns << %r{/gems/[0-9][^/]+/gems/}
 end
+
+#
+# So tests can clear class-level caches between examples.
+# TODO: Add this to the azure-armrest gem.
+#
+class Azure::Armrest::ArmrestService
+  def self.clear_caches
+    @@providers_hash = {}
+    @@tokens         = {}
+    @@subscriptions  = {}
+  end
+end
```

Comment:
@chrisarcand 
Just to make sure I'm not misunderstanding. When you say: _There isn't a good way to load it on a case by case basis_, you mean load the test macro file(s), right? like via `require`? I was thinking that only the spec files that need the test macros would require them. However, that would be a moot point if we place them in `gems/pending/spec/support`, because all of those files get required before the specs are run.